packages/vite-plugin-cloudflare/src/assets.ts

```diff
@@ -0,0 +1,18 @@
+import type { Miniflare } from 'miniflare';
+
+export interface AssetConfig {
+	htmlHandling?:
+		| 'auto-trailing-slash'
+		| 'force-trailing-slash'
+		| 'drop-trailing-slash'
+		| 'none';
+	notFoundHandling?: 'single-page-application' | '404-page' | 'none';
+}
+
+export const ROUTER_WORKER_NAME = '__router-worker__';
+export const ASSET_WORKER_NAME = '__asset-worker__';
+export const ASSET_WORKERS_COMPATIBILITY_DATE = '2024-10-04';
+
+export function getRouterWorker(miniflare: Miniflare) {
+	return miniflare.getWorker(ROUTER_WORKER_NAME);
```

Comment:
So this implementation does not support multiple user workers having assets.
I think this is actually a real requirement going forward, even if we don't land it in this iteration.
I can be onboard with the idea that in most (all?) scenarios there is only one entry-point Worker that hosts assets (from the point of view of the Vite HTTP server).

But we will need to support servicing binding to Workers with Assets from the entry-point Worker.

Reply:
This PR is very much about initial assets support to cover the most common use cases and meet the expectations of people used to using Vite.

Not ruling out multiple workers having assets at all but there are quite a few questions to answer before we do that and it will need its own dedicated PR.